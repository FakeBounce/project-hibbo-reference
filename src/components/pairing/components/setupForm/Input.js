import React from 'react';
import { Animated, TextInput } from 'react-native';
import {
  string,
  func,
  bool,
  object,
  array,
  number,
  oneOfType,
  shape,
  instanceOf,
} from 'prop-types';
import styled from 'styled-components/native';

import appStyles from 'styles/appStyles';

const SBorder = styled(Animated.View)`
  border-bottom-width: 2px;
  border-bottom-color: ${props => props.theme.colors.white};
  margin-horizontal: 20px;
  overflow: hidden;
`;
const SInput = styled(Animated.createAnimatedComponent(TextInput))`
  height: 45px;
  padding: 5px;
  font-size: ${props => props.theme.textSizes.regular};
  text-align: center;
  font-family: ${props => props.theme.fonts.circularBold};
`;

const Input = ({
  name,
  defaultValue,
  placeholder,
  setData,
  validate,
  error,
  secureEntry,
  keyboardType,
  capitalize,
  style,
  returnKeyType,
  maxLength,
  innerRef,
  animations,
  autoFocus,
}) => {
  return (
    <SBorder
      style={[
        style,
        {
          opacity: animations.borderOpacity,
          transform: [{ translateY: animations.borderTranslateY }],
        },
      ]}
    >
      <SInput
        key={keyboardType}
        value={defaultValue}
        autoCapitalize={capitalize}
        autoCorrect={false}
        autoFocus={autoFocus}
        keyboardType={keyboardType}
        onChangeText={text => {
          setData(name, text);
        }}
        innerRef={innerRef}
        onSubmitEditing={() => {
          validate(name);
        }}
        placeholder={placeholder}
        placeholderTextColor={
          error ? appStyles.colors.google : appStyles.colors.white
        }
        returnKeyType={returnKeyType}
        style={[
          style,
          {
            color: error ? appStyles.colors.google : appStyles.colors.white,
            opacity: animations.inputOpacity,
            transform: [{ translateY: animations.inputTranslateY }],
          },
        ]}
        secureTextEntry={secureEntry}
        underlineColorAndroid="transparent"
        maxLength={maxLength}
      />
    </SBorder>
  );
};

Input.defaultProps = {
  capitalize: 'words',
  style: {},
  returnKeyType: 'send',
  secureEntry: false,
  keyboardType: 'default',
  maxLength: 100,
  innerRef: null,
  autoFocus: false,
};

Input.propTypes = {
  innerRef: func,
  maxLength: number,
  capitalize: string,
  name: string.isRequired,
  defaultValue: string.isRequired,
  placeholder: string.isRequired,
  setData: func.isRequired,
  validate: func.isRequired,
  error: bool.isRequired,
  secureEntry: bool,
  keyboardType: string,
  style: oneOfType([object, number, array]),
  returnKeyType: string,
  animations: shape({
    borderOpacity: instanceOf(Animated.Value).isRequired,
    borderTranslateY: instanceOf(Animated.Value).isRequired,
    inputOpacity: instanceOf(Animated.Value).isRequired,
    inputTranslateY: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  autoFocus: bool,
};

export default Input;
