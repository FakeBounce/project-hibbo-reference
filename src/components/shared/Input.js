import React from 'react';
import { Dimensions } from 'react-native';
import { string, func, bool, oneOfType, object, number } from 'prop-types';

import theme from 'styles/appStyles';

import * as Styles from './styles';

const { width } = Dimensions.get('window');
const Input = ({
  name,
  keyboardType,
  capitalize,
  onChange,
  innerRef,
  onSubmit,
  placeholder,
  error,
  returnKeyType,
  style,
  secureEntry,
  maxLength,
  centered,
  value,
  inputWidth,
}) => {
  return (
    <Styles.SInputContainer centered={centered}>
      <Styles.SInput
        key={keyboardType}
        value={value}
        autoCapitalize={capitalize}
        autoCorrect={false}
        keyboardType={keyboardType}
        onChangeText={text => {
          onChange(name, text);
        }}
        innerRef={innerRef}
        onSubmitEditing={() => {
          onSubmit(name);
        }}
        placeholder={placeholder}
        placeholderTextColor={
          error ? theme.colors.google : theme.colors.inputGrey
        }
        returnKeyType={returnKeyType}
        style={style}
        inputWidth={inputWidth}
        secureTextEntry={secureEntry}
        underlineColorAndroid="transparent"
        maxLength={maxLength}
      />
      <Styles.SBorderInput inputWidth={inputWidth} />
    </Styles.SInputContainer>
  );
};

Input.defaultProps = {
  keyboardType: 'default',
  capitalize: 'words',
  innerRef: null,
  returnKeyType: 'next',
  style: {},
  secureEntry: false,
  maxLength: 500,
  centered: false,
  inputWidth: width,
  onSubmit: () => {},
};

Input.propTypes = {
  name: string.isRequired,
  keyboardType: string,
  capitalize: string,
  onChange: func.isRequired,
  innerRef: func,
  onSubmit: func,
  placeholder: string.isRequired,
  error: bool.isRequired,
  returnKeyType: string,
  style: oneOfType([string, number, object]),
  secureEntry: bool,
  maxLength: number,
  centered: bool,
  value: oneOfType([string, number]).isRequired,
  inputWidth: number,
};

export default Input;
