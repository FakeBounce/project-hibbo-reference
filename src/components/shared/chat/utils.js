import dateFormat from 'dateformat';

const DEPRECATION_MESSAGE =
  'isSameUser and isSameDay should be imported from the utils module instead of using the props functions';

export const isSameDay = (currentMessage = {}, diffMessage = {}) => {
  if (!currentMessage.createdAt || !currentMessage.createdAt) {
    return false;
  }
  try {
    const currentCreatedAt = dateFormat(currentMessage.createdAt, 'd/m/yyyy');
    const diffCreatedAt = dateFormat(diffMessage.createdAt, 'd/m/yyyy');
    return currentCreatedAt === diffCreatedAt;
  } catch (err) {
    return false;
  }
};

export const isSameUser = (currentMessage = {}, diffMessage = {}) => {
  return Boolean(
    diffMessage.user &&
      currentMessage.user &&
      diffMessage.user._id === currentMessage.user._id,
  );
};

export const warnDeprecated = fn => {
  return (...args) => {
    console.warn(DEPRECATION_MESSAGE); // eslint-disable-line
    return fn(...args);
  };
};
