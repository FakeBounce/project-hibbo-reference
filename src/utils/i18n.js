import { Platform, NativeModules } from 'react-native';

import messages from '../i18n';

const startsWithVowel = param => {
  const letter = param[0];
  if (
    letter === 'a' ||
    letter === 'e' ||
    letter === 'i' ||
    letter === 'o' ||
    letter === 'u' ||
    letter === 'y'
  ) {
    return true;
  }

  return false;
};

export const getLanguage = () => {
  if (Platform.OS === 'ios' && NativeModules.SettingsManager) {
    if (
      NativeModules.SettingsManager.settings.AppleLocale.substr(0, 2) === 'fr'
    ) {
      return 'fr';
    }
  } else if (Platform.OS === 'android' && NativeModules.I18nManager) {
    if (NativeModules.I18nManager.localeIdentifier.substr(0, 2) === 'fr') {
      return 'fr';
    }
  }
  return 'en';
};

const getTrad = (target, key, defaultKey) => {
  return target[key] || defaultKey;
};

export const getTranslations = (key, params, elisionParam = null) => {
  const lang = getLanguage();
  let newKey = key;

  // Handle elision of the translation depending on a param
  if (elisionParam && params[elisionParam]) {
    newKey = startsWithVowel(params[elisionParam])
      ? `${key}.vowel`
      : `${key}.consonant`;
  }

  // Proxy doesn't work on old version of IOS and android
  // const i18n = new Proxy(messages[lang], {
  //   get(target, k) {
  //     return target[k] || key;
  //   },
  // });
  const res = getTrad(messages[lang], newKey, key);

  if (!res) return res;
  return res.replace(/\/%(.*?)%\//g, (matching, param) => params[param]);
};

export const getTrueNumber = (amount, type = '', currency = 'â‚¬') => {
  const absAmount = Math.abs(amount);
  const strAmount = absAmount.toString();
  let sign = ' ';

  if (type !== '') {
    sign = type === 'credit' ? '+' : '-';
  }

  if (strAmount === '0') {
    return ` ${strAmount}${currency}`;
  }
  if (strAmount.length <= 2) {
    return `${sign}0,${strAmount}${currency}`;
  }

  const fistDigits = strAmount.substring(0, strAmount.length - 2);
  const lastDigits = strAmount.substring(strAmount.length - 2);

  if (lastDigits === '00') {
    return `${sign}${fistDigits}${currency}`;
  }
  return `${sign}${fistDigits},${lastDigits}${currency}`;
};
