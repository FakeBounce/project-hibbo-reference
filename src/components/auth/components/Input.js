import React, { PureComponent } from 'react';
import {
  string,
  func,
  shape,
  bool,
  oneOfType,
  object,
  number,
  array,
} from 'prop-types';
import { getTranslations } from 'utils/i18n';
import appStyles from 'styles/appStyles';

import { SContainerInput, SInput } from '../styles';

class Input extends PureComponent {
  constructor(props) {
    super(props);
    this.inputText = null;
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.label !== nextProps.label) {
      this.inputText.focus();
    }
  }

  getKeyboardType = label => {
    switch (label) {
      case getTranslations('auth.label.email'):
        return 'email-address';
      case getTranslations('auth.label.phone'):
        return 'phone-pad';
      default:
        return 'default';
    }
  };

  getReturnKeyType = label => {
    if (label === getTranslations('auth.label.confirmpassword')) return 'done';
    return 'next';
  };

  getAutoCapitalize = label => {
    if (
      label === getTranslations('auth.label.firstname') ||
      label === getTranslations('auth.label.lastname')
    )
      return 'sentences';
    return 'none';
  };

  getResChange(text, keyboardType) {
    let res = text;
    if (keyboardType !== 'default') {
      res = text.replace(/\s/g, '');
    }
    if (keyboardType === 'phone-pad') {
      res = res.replace(/[-|.|(|)]/g, '');
    }
    return res;
  }

  needSecurity(label) {
    return (
      label === getTranslations('auth.label.password') ||
      label === getTranslations('auth.label.confirmpassword')
    );
  }

  render() {
    const {
      form,
      label,
      name,
      onChange,
      validate,
      error,
      title,
      hide,
      autoFocus,
      inputTextSize,
      inputStyle,
    } = this.props;
    const keyboardType = this.getKeyboardType(label);

    return (
      <SContainerInput>
        <SInput
          key={keyboardType}
          innerRef={ref => {
            this.inputText = ref;
          }}
          value={form[name]}
          autoCapitalize={this.getAutoCapitalize(label)}
          autoCorrect={false}
          autoFocus={autoFocus}
          keyboardType={keyboardType}
          onChangeText={text => {
            onChange(name, this.getResChange(text, keyboardType));
          }}
          onSubmitEditing={() => validate(title)}
          placeholder={label}
          placeholderTextColor={
            error ? appStyles.red : appStyles.toggleButtonGrey
          }
          returnKeyType={this.getReturnKeyType(label)}
          secureTextEntry={this.needSecurity(label) && hide}
          inputTextSize={inputTextSize}
          style={inputStyle}
          underlineColorAndroid="transparent"
        />
      </SContainerInput>
    );
  }
}

Input.defaultProps = {
  hide: false,
  inputStyle: {
    height: 75,
    marginTop: 90,
  },
  autoFocus: true,
};

Input.propTypes = {
  inputStyle: oneOfType([object, number, array]),
  hide: bool,
  autoFocus: bool,
  name: string.isRequired,
  label: string.isRequired,
  onChange: func.isRequired,
  validate: func.isRequired,
  form: shape({
    email: string,
    firstName: string,
    lastName: string,
    pwd: string,
  }).isRequired,
  error: bool.isRequired,
  title: string.isRequired,
  inputTextSize: number.isRequired,
};

export default Input;
