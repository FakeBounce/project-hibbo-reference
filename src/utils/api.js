import Config from 'react-native-config';
import { getLanguage } from 'utils/i18n';
import { Observable } from 'rxjs';
import { store } from 'index';
import { REFRESH_TOKEN } from 'actionTypes/userActionTypes';

export const API = Config.api;
const apiTimeout = 10000;
export const debounceTime = 400;
export const { mangoPayApi, apiKey } = Config;
const defaultHeaders = {
  Accept: 'application/json',
  'Content-Type': 'application/json',
  'Accept-language': getLanguage(),
};

const checkStatus = res => {
  if (res.status >= 200 && res.status < 300) {
    return res;
  }

  throw res;
};

const getRootUrl = (url, accountId) => {
  if (!accountId) return url;

  return `${url}/${accountId}`;
};

const parseJSON = res => {
  const token = res.headers.get('authorization');
  if (token && token !== store.getState().user.token) {
    store.dispatch({ type: REFRESH_TOKEN, payload: { token } });
  }
  return res.json();
};

export const getEpic = (url, defaultToken = '', accountId = '') => {
  const state = store.getState();
  const token =
    state && state.user && state.user.token ? state.user.token : defaultToken;

  const request = fetch(
    `${API}${getRootUrl(
      url,
      accountId !== '' ? accountId : state.user.userDatas.id,
    )}`,
    {
      headers: {
        Authorization: `Bearer ${token}`,
        ...defaultHeaders,
      },
      method: 'GET',
    },
  )
    .then(res => checkStatus(res))
    .then(data => parseJSON(data));

  return Observable.fromPromise(request)
    .do(data => console.log('Get res', data))
    .timeout(apiTimeout);
};

export function get(url, token = '') {
  return fetch(`${API}${url}`, {
    headers: {
      Authorization: `Bearer ${token}`,
      ...defaultHeaders,
    },
    method: 'GET',
  })
    .then(res => {
      return checkStatus(res);
    })
    .then(checkedRes => {
      return parseJSON(checkedRes);
    })
    .then(data => {
      console.log('Get res', data);
      return data;
    })
    .catch(err => {
      console.log('Get err', err);
    });
}

export const postEpic = (url, body) => {
  console.log('Post content', body);
  const state = store.getState();
  const token = state && state.user ? state.user.token : '';

  const request = fetch(
    `${API}${getRootUrl(url, !token ? '' : state.user.userDatas.id)}`,
    {
      headers: {
        Authorization: `Bearer ${token}`,
        ...defaultHeaders,
      },
      method: 'POST',
      body: JSON.stringify(body),
    },
  )
    .then(res => checkStatus(res))
    .then(data => parseJSON(data));

  return Observable.fromPromise(request)
    .do(data => console.log('Post res', data))
    .timeout(apiTimeout);
};

export async function post(url, body, token = '') {
  console.log('DEPRECATED !!!');
  console.log('Post content', body);

  try {
    const res = await fetch(`${API}${url}`, {
      headers: {
        Authorization: `Bearer ${token}`,
        ...defaultHeaders,
      },
      method: 'POST',
      body: JSON.stringify(body),
    });
    const checkedRes = await checkStatus(res);
    const data = await parseJSON(checkedRes);

    console.log('Post res', data);
    return data;
  } catch (error) {
    console.log('Post error', error);
    throw error;
  }
}

export const putEpic = (url, body) => {
  const state = store.getState();
  const token = state && state.user ? state.user.token : '';
  console.log('Put body', body);

  const request = fetch(
    `${API}${getRootUrl(url, !token ? '' : state.user.userDatas.id)}`,
    {
      headers: {
        Authorization: `Bearer ${token}`,
        ...defaultHeaders,
      },
      method: 'PUT',
      body: JSON.stringify(body),
    },
  )
    .then(res => checkStatus(res))
    .then(data => parseJSON(data));

  return Observable.fromPromise(request)
    .do(data => console.log('Put res', data))
    .timeout(apiTimeout);
};

export async function put(url, body, token = '') {
  console.log('DEPRECATED !!!');
  console.log('Put body', body);

  try {
    const res = await fetch(`${API}${url}`, {
      headers: {
        Authorization: `Bearer ${token}`,
        ...defaultHeaders,
      },
      method: 'PUT',
      body: JSON.stringify(body),
    });
    const checkedRes = await checkStatus(res);
    const data = await parseJSON(checkedRes);

    console.log('Put res', data);
    return data;
  } catch (error) {
    console.log('Put error', error);
    throw error;
  }
}

export const delEpic = url => {
  const state = store.getState();
  const token = state && state.user ? state.user.token : '';
  const request = fetch(
    `${API}${getRootUrl(url, !token ? '' : state.user.userDatas.id)}`,
    {
      headers: {
        Authorization: `Bearer ${token}`,
        ...defaultHeaders,
      },
      method: 'DELETE',
    },
  )
    .then(res => checkStatus(res))
    .then(checkedRes => parseJSON(checkedRes));

  return Observable.fromPromise(request)
    .do(data => console.log('Del res', data))
    .timeout(apiTimeout);
};

export async function del(url, token = '') {
  console.log('DEPRECATED !!!');
  try {
    const res = await fetch(`${API}${url}`, {
      headers: {
        Authorization: `Bearer ${token}`,
        ...defaultHeaders,
      },
      method: 'DELETE',
    });
    const checkedRes = await checkStatus(res);
    const data = await parseJSON(checkedRes);

    console.log('Del res', data);
    return data;
  } catch (error) {
    console.log('Del error', error);
    throw error;
  }
}
