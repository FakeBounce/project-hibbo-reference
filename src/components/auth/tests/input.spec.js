import { TextInput } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { getTranslations } from 'utils/i18n';
import Input from '../components/Input';

describe('Input', () => {
  const form = {
    email: '',
    firstName: '',
    lastName: '',
    pwd: '',
  };

  it('should render the component', () => {
    const fn = jest.fn();
    const component = renderer
      .create(
        <Input form={form} title="" label="" name="" onChange={fn} validate={fn} error={false} />,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should handle text change', () => {
    const fn = jest.fn();
    const fn2 = jest.fn();
    const tree = shallow(
      <Input form={form} title="" label="" name="" onChange={fn} validate={fn2} error={false} />,
    );

    tree.find(TextInput).first().simulate('ChangeText');
    expect(fn).toHaveBeenCalled();
  });

  it('should handle submit editing', () => {
    const fn = jest.fn();
    const fn2 = jest.fn();
    const tree = shallow(
      <Input form={form} title="" label="" name="" onChange={fn2} validate={fn} error={false} />,
    );

    tree.find(TextInput).first().simulate('SubmitEditing');
    expect(fn).toHaveBeenCalled();
  });

  it('should have a red placeholder if error', () => {
    const fn = jest.fn();
    const tree = shallow(
      <Input form={form} title="" label="" name="" onChange={fn} validate={fn} error />,
    );

    expect(tree.find(TextInput).get(0).props.placeholderTextColor).toEqual('#D5492F');
    expect(tree.find(TextInput).get(0).props.style).toEqual([undefined, { color: '#D5492F' }]);
  });

  it('should have autoCapitalize if label is "PrÃ©nom"', () => {
    const fn = jest.fn();
    const tree = shallow(
      <Input
        form={form}
        title=""
        label={getTranslations('auth.label.firstname')}
        name=""
        onChange={fn}
        validate={fn}
        error={false}
      />,
    );

    expect(tree.find(TextInput).get(0).props.autoCapitalize).toEqual('sentences');
  });

  it('should have autoCapitalize if label is "Nom"', () => {
    const fn = jest.fn();
    const tree = shallow(
      <Input
        form={form}
        title=""
        label={getTranslations('auth.label.lastname')}
        name=""
        onChange={fn}
        validate={fn}
        error={false}
      />,
    );

    expect(tree.find(TextInput).get(0).props.autoCapitalize).toEqual('sentences');
  });

  it('should have keyboardType of email if label is "Mail"', () => {
    const fn = jest.fn();
    const tree = shallow(
      <Input
        form={form}
        title=""
        label={getTranslations('auth.label.mail')}
        name=""
        onChange={fn}
        validate={fn}
        error={false}
      />,
    );

    expect(tree.find(TextInput).get(0).props.keyboardType).toEqual('email-address');
  });

  it('should have keyboardType of default if label is not "Mail"', () => {
    const fn = jest.fn();
    const tree = shallow(
      <Input form={form} title="" label="" name="" onChange={fn} validate={fn} error={false} />,
    );

    expect(tree.find(TextInput).get(0).props.keyboardType).toEqual('default');
  });

  it('should have returnKeyType of done if label is "Mot de passe"', () => {
    const fn = jest.fn();
    const tree = shallow(
      <Input
        form={form}
        title=""
        label={getTranslations('auth.label.password')}
        name=""
        onChange={fn}
        validate={fn}
        error={false}
      />,
    );

    expect(tree.find(TextInput).get(0).props.returnKeyType).toEqual('done');
  });

  it('should have returnKeyType of next if label is not "Mot de passe"', () => {
    const fn = jest.fn();
    const tree = shallow(
      <Input form={form} title="" label="" name="" onChange={fn} validate={fn} error={false} />,
    );

    expect(tree.find(TextInput).get(0).props.returnKeyType).toEqual('next');
  });

  it('should have secureTextEntry if label is "Mot de passe"', () => {
    const fn = jest.fn();
    const tree = shallow(
      <Input
        form={form}
        title=""
        label={getTranslations('auth.label.password')}
        name=""
        onChange={fn}
        validate={fn}
        error={false}
      />,
    );

    expect(tree.find(TextInput).get(0).props.secureTextEntry).toEqual(true);
  });

  it('should not have secureTextEntry if label is not "Mot de passe"', () => {
    const fn = jest.fn();
    const tree = shallow(
      <Input form={form} title="" label="" name="" onChange={fn} validate={fn} error={false} />,
    );

    expect(tree.find(TextInput).get(0).props.secureTextEntry).toEqual(false);
  });
});
