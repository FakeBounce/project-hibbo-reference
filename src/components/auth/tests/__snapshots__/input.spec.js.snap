// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render the component 1`] = `
<View
  style={undefined}
>
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    autoFocus={true}
    keyboardType="default"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder=""
    placeholderTextColor={undefined}
    returnKeyType="next"
    secureTextEntry={false}
    style={
      Array [
        undefined,
        Object {
          "color": undefined,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value={undefined}
  />
</View>
`;
